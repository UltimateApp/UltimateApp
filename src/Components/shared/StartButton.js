import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';

import theme from '../../styles/theme.style';

const StartButton = props => {
  return (
    <TouchableOpacity onPress={props.onPress} testID="startButton" style={styles.button}>
      <Text style={styles.text}>{props.text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    margin: 20,
    width: 100,
    height: 100,
    borderRadius: 50,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.MAIN_COLOR,
  },
  text: {
    fontSize: theme.FONT_SIZE_MEDIUM,
    fontWeight: 'bold',
    color: theme.COLOR_PRIMARY_LIGHT,
  },
});

export default StartButton;
