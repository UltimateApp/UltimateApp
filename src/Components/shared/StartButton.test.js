import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from 'react-native-testing-library';

import StartButton from './StartButton';

afterEach(cleanup);

describe('<StartButton />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<StartButton text="OK" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls delegates onPress prop', async () => {
    const onPress = jest.fn();

    const { getByTestId } = render(<StartButton text="Easy" onPress={onPress} />);

    await fireEvent.press(getByTestId('startButton'));

    expect(onPress).toHaveBeenCalled();
  });
});
