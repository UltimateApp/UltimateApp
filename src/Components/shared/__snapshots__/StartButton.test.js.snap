// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#08AEEA",
      "borderRadius": 50,
      "height": 100,
      "justifyContent": "center",
      "margin": 20,
      "opacity": 1,
      "width": 100,
    }
  }
  testID="startButton"
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    OK
  </Text>
</View>
`;
